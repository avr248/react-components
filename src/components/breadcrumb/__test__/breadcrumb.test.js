import React from 'react';
import renderer from 'react-test-renderer';
import Breadcrumb from '..';

describe('Breadcrumb component', () => {
  describe('Breadcrumb.Item component', () => {
    it('should render <li>', () => {
      const component = renderer.create(
        <Breadcrumb.Item>item</Breadcrumb.Item>,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('should have is-active class', () => {
      const component = renderer.create(
        <Breadcrumb.Item active>active item</Breadcrumb.Item>,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  it('Should be a Breadcrumb', () => {
    const component = renderer.create(
      <Breadcrumb>
        <Breadcrumb.Item>
          <a href="/">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="/section">Section</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="/detail">Details</a>
        </Breadcrumb.Item>
      </Breadcrumb>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  [null, 'arrow', 'dot', 'bullet', 'succeeds'].map((separator) =>
    it(`should use separator ${separator}`, () => {
      const component = renderer.create(
        <Breadcrumb separator={separator}>
          <Breadcrumb.Item>
            <a href="#1">Storybook</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="#2">Breadcrumb</a>
          </Breadcrumb.Item>
          <Breadcrumb.Item>
            <a href="#3">Breadcrumb Types</a>
          </Breadcrumb.Item>
        </Breadcrumb>,
      );
      expect(component.toJSON()).toMatchSnapshot();
    }),
  );

  it('Should use inline style and custom size', () => {
    const component = renderer.create(
      <Breadcrumb style={{ marginTop: 10 }} size="large">
        <Breadcrumb.Item>
          <a href="/">Home</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item>
          <a href="/section">Section</a>
        </Breadcrumb.Item>
        <Breadcrumb.Item active>
          <a href="/detail">Details</a>
        </Breadcrumb.Item>
      </Breadcrumb>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
