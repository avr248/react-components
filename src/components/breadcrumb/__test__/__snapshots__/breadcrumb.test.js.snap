// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component Breadcrumb.Item component should have is-active class 1`] = `
<li
  className="is-active"
>
  active item
</li>
`;

exports[`Breadcrumb component Breadcrumb.Item component should render <li> 1`] = `
<li>
  item
</li>
`;

exports[`Breadcrumb component Should be a Breadcrumb 1`] = `
<nav
  className="breadcrumb"
>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/section"
      >
        Section
      </a>
    </li>
    <li>
      <a
        href="/detail"
      >
        Details
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component Should use inline style and custom size 1`] = `
<nav
  className="breadcrumb is-large"
  style={
    Object {
      "marginTop": 10,
    }
  }
>
  <ul>
    <li>
      <a
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/section"
      >
        Section
      </a>
    </li>
    <li
      className="is-active"
    >
      <a
        href="/detail"
      >
        Details
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component should use separator arrow 1`] = `
<nav
  className="breadcrumb has-arrow-separator"
>
  <ul>
    <li>
      <a
        href="#1"
      >
        Storybook
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        Breadcrumb
      </a>
    </li>
    <li>
      <a
        href="#3"
      >
        Breadcrumb Types
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component should use separator bullet 1`] = `
<nav
  className="breadcrumb has-bullet-separator"
>
  <ul>
    <li>
      <a
        href="#1"
      >
        Storybook
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        Breadcrumb
      </a>
    </li>
    <li>
      <a
        href="#3"
      >
        Breadcrumb Types
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component should use separator dot 1`] = `
<nav
  className="breadcrumb has-dot-separator"
>
  <ul>
    <li>
      <a
        href="#1"
      >
        Storybook
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        Breadcrumb
      </a>
    </li>
    <li>
      <a
        href="#3"
      >
        Breadcrumb Types
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component should use separator null 1`] = `
<nav
  className="breadcrumb"
>
  <ul>
    <li>
      <a
        href="#1"
      >
        Storybook
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        Breadcrumb
      </a>
    </li>
    <li>
      <a
        href="#3"
      >
        Breadcrumb Types
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Breadcrumb component should use separator succeeds 1`] = `
<nav
  className="breadcrumb has-succeeds-separator"
>
  <ul>
    <li>
      <a
        href="#1"
      >
        Storybook
      </a>
    </li>
    <li>
      <a
        href="#2"
      >
        Breadcrumb
      </a>
    </li>
    <li>
      <a
        href="#3"
      >
        Breadcrumb Types
      </a>
    </li>
  </ul>
</nav>
`;
